src/charm.py

```diff
@@ -4,24 +4,50 @@
 
 """Charmed Machine Operator for MySQL."""
 
+import hashlib
 import logging
+import secrets
+import string
 
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
 
-from mysqlsh_helpers import MySQL
+from mysqlsh_helpers import (
+    MySQL,
+    MySQLConfigureInstanceError,
+    MySQLConfigureMySQLUsersError,
+)
 
 logger = logging.getLogger(__name__)
 
+PASSWORD_LENGTH = 24
+PEER = "mysql-replicas"
```

Comment:
read from meta file?

Reply:
I say, keep on code, which is what everyone else is doing.